packages/@expo/config-plugins/src/utils/generateCode.ts

```diff
@@ -10,8 +10,8 @@ function getGeneratedSectionIndexes(
   tag: string
 ): { contents: string[]; start: number; end: number } {
   const contents = src.split('\n');
-  const start = contents.findIndex((line) => line.includes(`@generated begin ${tag}`));
-  const end = contents.findIndex((line) => line.includes(`@generated end ${tag}`));
+  const start = contents.findIndex((line) => RegExp(`@generated begin ${tag} -`).test(line));
+  const end = contents.findIndex((line) => RegExp(`@generated end ${tag}$`).test(line));
```

Comment:
```suggestion
  const end = contents.findIndex((line) => new RegExp(`@generated end ${tag}$`).test(line));
```

Please use `new` to construct the object.